### PR TITLE
Use variable_data() in tensor_to_numpy

### PR DESCRIPTION
**Pull request by yf225**
As part of the Variable/Tensor merge, we want to gradually remove call sites of `tensor_data()` and the API itself, and instead uses `variable_data()`. This PR removes the `tensor_data()` call in the tensor_to_numpy conversion path.

### CODE REVIEW

File: torch/csrc/utils/tensor_numpy.cpp [hunk 1]
```diff
@@ -110,7 +115,7 @@ PyObject* tensor_to_numpy(const at::Tensor& tensor) {
   // object of the ndarray to the tensor and disabling resizes on the storage.
   // This is not sufficient. For example, the tensor's storage may be changed
   // via Tensor.set_, which can free the underlying memory.
-  PyObject* py_tensor = THPVariable_Wrap(make_variable(tensor, false));
+  PyObject* py_tensor = THPVariable_Wrap(as_variable_ref(tensor).variable_data());
```
Comment (yf225) [inline torch/csrc/utils/tensor_numpy.cpp]: We could add an explicit check for `tensor.is_variable()` here, but I think `as_variable_ref(tensor)` already checks whether `tensor` is a Variable.

Comment (gchanan) [reply torch/csrc/utils/tensor_numpy.cpp]: Do you need the `variable_data`?

Comment (yf225) [reply torch/csrc/utils/tensor_numpy.cpp]: I just checked and we don't need to use `variable_data` here - resizing is not supported on either the PyTorch tensor or the numpy array, so creating a shallow-copy of the tensor being converted is not necessary in any case.

**Review by gchanan — APPROVED** [review]
nice.

File: torch/csrc/utils/tensor_numpy.cpp [hunk 2]
```diff
@@ -85,6 +85,11 @@ PyObject* tensor_to_numpy(const at::Tensor& tensor) {
   if (tensor.type().backend() != Backend::CPU) {
     throw TypeError("NumPy conversion for %s is not supported", tensor.type().toString().c_str());
   }
+  if (tensor.requires_grad()) {
+    throw std::runtime_error(
+        "Can't call numpy() on Variable that requires grad. "
```
Comment (Atcold) [inline torch/csrc/utils/tensor_numpy.cpp]: This error message is outdated. There are no more `Variables` in PyTorch.

Comment (gchanan) [reply torch/csrc/utils/tensor_numpy.cpp]: There are in C++ still, see https://github.com/pytorch/pytorch/issues/13638.  We should fix these once that proposal is implemented, though.

**Pull request closed without merging**